src/textual/widgets/_selection_list.py

```diff
@@ -0,0 +1,655 @@
+"""Provides a selection list widget, allowing one or more items to be selected."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Callable, ClassVar, Generic, Iterable, TypeVar, cast
+
+from rich.repr import Result
+from rich.segment import Segment
+from rich.style import Style
+from rich.text import Text, TextType
+from typing_extensions import Self
+
+from ..binding import Binding
+from ..messages import Message
+from ..strip import Strip
+from ._option_list import NewOptionListContent, Option, OptionList
+from ._toggle_button import ToggleButton
+
+SelectionType = TypeVar("SelectionType")
+"""The type for the value of a [`Selection`][textual.widgets.selection_list.Selection] in a [`SelectionList`][textual.widgets.SelectionList]"""
+
+MessageSelectionType = TypeVar("MessageSelectionType")
+"""The type for the value of a [`Selection`][textual.widgets.selection_list.Selection] in a [`SelectionList`][textual.widgets.SelectionList] message."""
+
+
+class SelectionError(TypeError):
+    """Type of an error raised if a selection is badly-formed."""
+
+
+class Selection(Generic[SelectionType], Option):
+    """A selection for a [`SelectionList`][textual.widgets.SelectionList]."""
+
+    def __init__(
+        self,
+        prompt: TextType,
+        value: SelectionType,
+        initial_state: bool = False,
+        id: str | None = None,
+        disabled: bool = False,
+    ):
+        """Initialise the selection.
+
+        Args:
+            prompt: The prompt for the selection.
+            value: The value for the selection.
+            initial_state: The initial selected state of the selection.
+            id: The optional ID for the selection.
+            disabled: The initial enabled/disabled state. Enabled by default.
+        """
+        if isinstance(prompt, str):
+            prompt = Text.from_markup(prompt)
+        super().__init__(prompt.split()[0], id, disabled)
```

Comment:
Will this blow up if `prompt` is empty string ?

Reply:
I don't believe so.

```python
>>> Text("").split()[0]

>>> 
```

Not the first time [this approach has been used in Textual](https://github.com/Textualize/textual/blob/8151946f380acacb8daa39b9c5b705033f5263d6/src/textual/widgets/_tree.py#L665-L669).

Reply:
Hmm. `Text.split` should have the same behaviour as `str.split`. May be a bug in Rich.

Reply:
Ah, I also thought this would blow up but then I checked it and a split on an empty text produces an empty line.